eval/bbh/run_eval.py

```diff
@@ -47,15 +54,14 @@ def eval_hf_model(args, model, tokenizer, examples, task_prompt, save_path=None)
     for example, output in zip(examples, outputs):
         example["raw_output"] = output
         
-        # only keep the first part of the output - this is mainly for vanilla language models.
-        output = output.strip().split("\n\n")[0].strip()
-
         # extract the first answer after `So the answer is` and before the next period.
         # if there is no such answer, we will just use the raw output.
         results = re.search(r"So the answer is (.*?)\.", output)
         if results:
             prediction = results.group(1).strip()
         else:
+            # only keep the first part of the output - this is mainly for vanilla language models.
+            output = output.strip().split("\n\n")[0].strip()
```

Comment:
Why this change? For the behaviour with Llama-2?

Reply:
Sometimes the "So the answer is" appear in the second part after "\n\n", so I think we can try to find the "So the answer is" first, and rollback to the first part of the answer if not found.

For LLaMa2-chat, it's more complicated. The model cannot follow the few-shot demonstration format. It cannot generate the "So the answer is" string, so its performance is still bad.